model-optimizer/mo/ops/pad.py

```diff
@@ -85,6 +86,13 @@ def infer(node):
         PermuteInputs().set_input_permutation(node.in_node(1), node, 'input:0', 'shape')
         PermuteInputs().set_input_permutation(node.in_node(2), node, 'input:0', 'shape')
 
+    @staticmethod
+    def reverse_infer(node: Node):
+        input_shape = node.in_port(0).data.get_shape()
+        if input_shape is None and node.is_in_port_connected(2) and node.in_port(2).data.get_shape() is not None:
+            shape = undefined_shape_of_rank(node.in_port(2).data.get_shape()[0])
```

Comment:
The third input to `Pad` is `pads_end`  and a number of elements there should be a rank. I think we need shape = undefined_shape_of_rank(len(shape)).
could you please clarify?

Reply:
The `len(node.in_port(2).data.get_shape())` and `node.in_port(2).data.get_shape()[0]` are equivalent in this case because according to the specification of [Pad](https://docs.openvino.ai/latest/openvino_docs_ops_movement_Pad_1.html): "2 : pads_begin 1D tensor of type T_INT. ". So the length of 1D tensor and the first element of its shape are always equal.

For example, if `pads_begin` is `[2, 3, 4, 5]` then `len(pads_begin) = 4` and `shape(pads_begin) = [4]` so `shape(pads_begin)[0] = 4`